### PR TITLE
[WIP] Send email notification after safety comments are changed

### LINKED ISSUE #245: Safety file comments
Send notification to dream manager email once safety manager add a comment in the safety field (welcome screen).
Save comments history.

### PR DESCRIPTION
**Pull request by enoodle**
Fixes #245 

Created a simpler mailer class to send a notification email to the user that edited the safety comments and the Dream (camp) creator.

The email  contents can be edited in : `app/views/camp_mailer/safety_comments_updated.html.erb` (html)
or `app/views/camp_mailer/safety_comments_updated.text.erb` (text)

@LioraNaginsky : The email is using the general email layout from `app/views/layouts/mailer.{html,text}.erb` - In those files there is a line `<%= yield %>` that inserts the specific mail content (from the above files) and other than that we can edit those files to insert common email content.

I also enabled the preview (assuming there is one camp and user in the system) at `http://localhost:3000/rails/mailers/camp_mailer/safety_comments_updated`

@LioraNaginsky @rootux

### CODE REVIEW

**Comment by rootux** [comment]
@enoodle Looks great. Can you specify which tests did you do to make sure this works?

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: "dream@midburn.org"
```
Comment (rootux) [inline app/mailers/application_mailer.rb]: @enoodle Should be "dreams@midburn.org" instead of dream

File: app/mailers/camp_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CampMailer < ApplicationMailer
+  default from: 'dreams@midburn.org'
+
+  def safety_comments_updated(camp, editor)
+    @camp = camp
+    @editor = editor
+    mail(:to      => [@camp.creator.email, @editor.email].uniq!,
+         :subject => "Safety Comment Updated For Dream #{camp.name}")
```
Comment (rootux) [inline app/mailers/camp_mailer.rb]: Should use i18n Translation english->hebrew

File: app/views/camp_mailer/safety_comments_updated.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h1>Dream Safety Comments Were Updated for <%= @camp.name %></h1>
```
Comment (rootux) [inline app/views/camp_mailer/safety_comments_updated.html.erb]: Should use i18n Translation english->hebrew

File: app/views/camp_mailer/safety_comments_updated.html.erb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    <p>
+      <%= @editor.email %> has updated the safety comments for this dream.<br>
+    </p>
+    <p>
+    <%= @camp.safety_file_comments %>
+    </p>
+    <p>SOME CLOSING LINE OR UNICORN!</p>
```
Comment (rootux) [inline app/views/camp_mailer/safety_comments_updated.html.erb]: Please replace with closing line or a unicorn :) Or remove

File: app/views/camp_mailer/safety_comments_updated.text.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Dream Safety Comments Were Updated for <%= @camp.name %>
+=======================================<%= '='*@camp.name.length %>
+
+<%= @editor.email %> has updated the safety comments for this dream:
+
+<%= @camp.safety_file_comments %>
+
+SOME CLOSING LINE OR UNICORN!
```
Comment (rootux) [inline app/views/camp_mailer/safety_comments_updated.text.erb]: Please remove or add a closing line

**Review by rootux — CHANGES_REQUESTED** [review]

**Comment by enoodle** [comment]
@rootux 
> Can you specify which tests did you do to make sure this works?

I added two tests: one that makes sure that the controller can render a simple sample mail, sanity check to prevent us from breaking something in the view files. Another is a test in the camps controller spec that makes sure that when the comment file is updated we are calling the email sending function in that controller.

**Pull request merged**